src/machine/machine_rp2040_i2c.go

```diff
@@ -57,6 +57,8 @@ var (
 	ErrInvalidTgtAddr     = errors.New("invalid target i2c address not in 0..0x80 or is reserved")
 	ErrI2CGeneric         = errors.New("i2c error")
 	ErrRP2040I2CDisable   = errors.New("i2c rp2040 peripheral timeout in disable")
+	errInvalidI2CSDA      = errors.New("invalid I2C SDA pin")
```

Comment:
These should be exported if the developer wants to compare against them for specific error handling, right?

Reply:
I'm not sure in what situation a user would want to handle an error regarding chosen pins. Seems like that would be always chosen as a constant literal in the program setup. Might it be unecessary noise to the machine package?

Reply:
Since all the other errors are exported, shouldn't we be consistent? But you are probably correct about the nature of the error.

Reply:
I'm unsure of the benefits of being consistent in this case. Every error we add here decreases the portability of the code since this is a board specific file.

There's also plenty of examples in the Go standard library where there's a distinction made between what should and should not be made an exported error. Keep in mind these are errors that could be exported in the future if the need ever arised, as has happened recently with the "deadline exceeded" error in the net package, I believe.
![image](https://user-images.githubusercontent.com/26156425/220185919-3972a5c0-14fc-41f4-a3a0-a4273f5d66b6.png)

Reply:
I think not exporting this error is fine. The only thing you can do with this error is show it to the developer so they can fix the code. It's not something the application should handle.

In addition, every thing that is not exported can be exported without breaking the API. Once it is exported, changing it to be non-exported is an API break. So starting out by making it unexported is always safe.